libyul/optimiser/Metrics.cpp

```diff
@@ -60,3 +62,62 @@ void CodeSize::visit(Expression const& _expression)
 	++m_size;
 	ASTWalker::visit(_expression);
 }
+
+
+size_t CodeCost::codeCost(Expression const& _expr)
+{
+	CodeCost cc;
+	cc.visit(_expr);
+	return cc.m_cost;
+}
+
+
+void CodeCost::operator()(FunctionCall const& _funCall)
+{
+	m_cost += 49;
+	ASTWalker::operator()(_funCall);
+}
+
+void CodeCost::operator()(FunctionalInstruction const& _instr)
+{
+	using namespace dev::solidity;
+	Tier gasPriceTier = instructionInfo(_instr.instruction).gasPriceTier;
+	if (gasPriceTier < Tier::VeryLow)
+		m_cost -= 1;
```

Comment:
Can this underflow?
If so, we should either have a ``if (m_cost > 0)`` guard or change ``m_cost`` to a signed type and check for ``cost <= 0`` instead of ``cost == 0`` below.
If not, we might want to have a ``m_cost > 0`` assertion here.

Reply:
It should not, because of https://github.com/ethereum/solidity/pull/5620/files#diff-368e3d884b54a1cec127b73fa1d8c9e3R95, but probably better to add the assert.

Reply:
Updated.